src/axolotl/prompt_tokenizers.py

```diff
@@ -402,18 +404,24 @@ def tokenize_prompt(self, prompt):
                         add_eos_token=False,
                         strip_bos_token=True,
                     )
-                    # not masked out from labels
                     labels = copy.deepcopy(res["input_ids"])
-                    len_role = len(role_res["input_ids"])
-                    labels[:len_role] = [IGNORE_TOKEN_ID] * min(len_role, len(labels))
+                    if not self.train_on_inputs:
+                        # mask out role tokens from the labels
+                        len_role = len(role_res["input_ids"])
+                        labels[:len_role] = [IGNORE_TOKEN_ID] * min(
+                            len_role, len(labels)
+                        )
```

Comment:
```suggestion
                        len_role = len(role_res["input_ids"])
                        labels[:len_role] = [IGNORE_TOKEN_ID] * len_role
```

Can I do this? I'm not sure why there was a `min`.

Reply:
this was introduced in https://github.com/OpenAccess-AI-Collective/axolotl/pull/723, I'd prefer to leave it for now and prevent a regression.